gcp_variant_transforms/beam_io/bgzf_io.py

```diff
@@ -0,0 +1,176 @@
+# Copyright 2019 Google Inc.  All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Splits the BGZF file."""
+
+import struct
+from collections import namedtuple
+from typing import Iterable, List, Tuple  # pylint: disable=unused-import
+
+from apache_beam.io import filesystem
+from apache_beam.io import filesystems
+
+# `Block` provides real file offset of the start and end of a gzip block. Given
+# it, one can directly seek to the start of the gzip block and decompress it.
+Block = namedtuple('Block', ['start', 'end'])
+
+_INDEX_FILE_APPENDIX = '.tbi'
+
+# `MAX_BLOCK_SIZE` defines the maximum amount of data included in one `Block`.
+MAX_BLOCK_SIZE = 1 * 1024 * 1024
+# It beyonds the normal bin range (1-37449), and contains other meta data rather
```

Comment:
`# This bin index and any beyond it contain index metadata and are not used to address blocks.`

At least that's my understanding (eg, 37451 is also not expected)